### DISCUSSION TITLE
Archival Strategy for Huge Data

### DISCUSSION THREAD

**Discussion by vakharia-deep** [ClickHouse/ClickHouse#50603] [Q&A] [answered]
We are using clickhouse on production env and as of now we have around 12 tbs data. Does clickhouse provide any data archival and retrieval mechanism out of the box , if not how do you archive data.

**Comment by DanRoscigno** [comment]
Hi @vakharia-deep , please have a look at the [TTL](https://clickhouse.com/docs/en/guides/developer/ttl) docs.

Half way down the page is a hot / warm / cold example.

**Reply by vakharia-deep** [reply]
Can I have my data to written to GCS bucket based on TTL rather then writing it to another disk and then load it on to the disk whenever needed?

**Comment by DanRoscigno** [comment, marked as answer]
You can use TTL to move the data to GCS, and you can also access the data on GCS whenever you need to--you do not have to copy it back to your ClickHouse system as ClickHouse can treat GCS as a disk. Here are details on mounting GCS buckets as a ClickHouse disk: https://clickhouse.com/docs/en/integrations/gcs#gcs-backed-mergetree

**Comment by DanRoscigno** [comment]
So, by combining the TTL and the GCS backed MergeTree, you could use local disk for speed, and then GCS to store older data to save money, and still access all of the data without having to move data back from GCS to local disk.

**Reply by vakharia-deep** [reply]
I'll have a look at, it appreciate the prompt response.
